### PR TITLE
Refactores the 'global search'

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**

### CODE REVIEW

**Comment by emanuelschuetze** [comment]
- sort motions by id (done by Finn)
- remove white BG
- fix overlapping motion titles

**Comment by FinnStutzenstein** [comment]
> * sort motions by id (done by Finn)

All entries are sorded by their displayed title, not (database) id.

**Comment by GabrielInTheWorld** [comment]
> * sort motions by id (done by Finn)

I've added a `mat-select` to have the option to sort by id or title.

File: client/src/app/core/ui-services/search.service.ts [hunk 1]
```diff
@@ -138,7 +138,7 @@ export class SearchService {
      * @param inCollectionStrings All connection strings which should be used for searching.
      * @returns All search results sorted by the model's title (via `getTItle()`).
      */
-    public search(query: string, inCollectionStrings: string[]): SearchResult[] {
+    public search(query: string, inCollectionStrings: string[], sortingProperty: string): SearchResult[] {
```
Comment (FinnStutzenstein) [inline client/src/app/core/ui-services/search.service.ts]: `sortingProperty`: 'id' | 'title'

File: client/src/app/site/common/components/search/search.component.html [hunk 2]
```diff
@@ -1,58 +1,72 @@
...
-</div>
 
-<mat-menu #menu="matMenu">
-    <button mat-menu-item *ngFor="let registeredModel of registeredModels" (click)="toggleModel(registeredModel)">
-        <mat-checkbox [checked]="registeredModel.enabled" (click)="$event.preventDefault()">
-            <span>{{ registeredModel.verboseNamePlural | translate }}</span>
-        </mat-checkbox>
-    </button>
-</mat-menu>
+        <mat-form-field class="search-sort">
```
Comment (tsiegleauq) [inline client/src/app/site/common/components/search/search.component.html]: looks a little bit odd. Better put either in the page-menu, which could be extended anyways.

- change `toggleModel` to pass a click-event, use `event.stopPropagation();` to leave the menu open after klick
- add `Sort by ID` and `Sort by title` to the menu.
- Use a `mat-divider` to separate the sorting from the rest of the content

File: client/src/app/site/common/components/search/search.component.html [hunk 3]
```diff
@@ -1,58 +1,72 @@
...
+        </button>
+    </mat-menu>
 
-<ng-container *ngIf="searchResultCount > 0">
-    <h3>
-        {{ searchResultCount }}
-        <span *ngIf="searchResultCount === 1" translate>result</span>
-        <span *ngIf="searchResultCount > 1" translate>results</span>
-    </h3>
+    <div *ngIf="searchResults.length > 0">
```
Comment (tsiegleauq) [inline client/src/app/site/common/components/search/search.component.html]: according to out mock-ups, the whole thing could be a `mat-card`, or something else with a plain white background.
The first entry could be a `mat-divider`, to separate the search from the results.

**Pull request merged**